src/controller/CHIPDeviceController.cpp

```diff
@@ -485,19 +468,9 @@ CHIP_ERROR DeviceController::UpdateDevice(Device * device, uint64_t fabricId)
 
 void DeviceController::PersistDevice(Device * device)
 {
-    // mStorageDelegate would not be null for a typical pairing scenario, as Pair()
-    // requires a valid storage delegate. However, test pairing usecase, that's used
-    // mainly by test applications, do not require a storage delegate. This is to
-    // reduce overheads on these tests.
-    // Let's make sure the delegate object is available before calling into it.
-    if (mStorageDelegate != nullptr && mState == State::Initialized)
+    if (mState == State::Initialized)
```

Comment:
please add a log if not initialized, to not silently fail persistence.

Ideally Persist would return a CHIP_ERROR. If I call "persistdevice" I do not expect it to potentially silently fail.

Reply:
So the Device does return an error but the controller code entirely ignores missing storage delegates (because it's missing only in tests). 

I've updated it to log errors from Persist() on the CHIPDevice and I've added a log if the state is not initialized